src/tests/infra.rs

```diff
@@ -0,0 +1,134 @@
+#[cfg(all(feature = "shuttle", test))]
+mod randomized {
+    use crate::ff::Fp32BitPrime;
+    use crate::helpers::Direction;
+    use crate::protocol::context::{Context, SemiHonestContext};
+    use crate::protocol::{QueryId, RecordId};
+    use crate::secret_sharing::Replicated;
+    use crate::test_fixture::Reconstruct;
+    use crate::test_fixture::{Runner, TestWorld};
+    use futures_util::future::{try_join, try_join_all};
+
+    #[test]
+    fn send_receive_sequential() {
+        shuttle::check_random(
+            || {
+                shuttle::future::block_on(async {
+                    let world = TestWorld::new(QueryId);
+                    let input = (0u32..100).map(Fp32BitPrime::from).collect::<Vec<_>>();
+
+                    let output = world
+                        .semi_honest(
+                            input.clone(),
+                            |ctx: SemiHonestContext<'_, Fp32BitPrime>, mut shares| async move {
+                                let (left_ctx, right_ctx) =
+                                    (ctx.narrow("left"), ctx.narrow("right"));
+                                let left_channel = left_ctx.mesh();
+                                let right_channel = right_ctx.mesh();
+                                let left_peer = ctx.role().peer(Direction::Left);
+                                let right_peer = ctx.role().peer(Direction::Right);
+
+                                // send all shares to the right peer
+                                for (i, share) in shares.iter().enumerate() {
+                                    let record_id = RecordId::from(i);
+                                    left_channel
+                                        .send(right_peer, record_id, share.left())
+                                        .await
+                                        .unwrap();
+                                    right_channel
+                                        .send(right_peer, record_id, share.right())
+                                        .await
+                                        .unwrap();
+                                }
```

Comment:
This seems exactly like `Reveal`. I'm not sure what you're doing here. Why not just call reveal?